src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -12,76 +23,78 @@
-            .name("Obi")
-            .gitHubID("Obinnabii")
-            .description("Obi is awesome.")
-            .interestTags(ObiSkills)
-            .projectIDs(projectIDs)
-            .buildPerson();
-    people.add(Obi);
-
-    ArrayList<String> SamiSkills = new ArrayList<String>();
-    SamiSkills.add("Stone carver");
-    SamiSkills.add("Bootstrap convert");
-    PersonObject Sami =
-        new PersonBuilder()
-            .name("Sami")
-            .gitHubID("Sami-2000")
-            .description("Sami is fun.")
-            .interestTags(SamiSkills)
-            .projectIDs(projectIDs)
-            .buildPerson();
-    people.add(Sami);
-
-    ArrayList<String> RichiSkills = new ArrayList<String>();
-    RichiSkills.add("Minecraft boss");
-    RichiSkills.add("React wizard");
-    PersonObject Richi =
-        new PersonBuilder()
-            .name("Richi")
-            .gitHubID("Richie78321")
-            .description("Richi is cool.")
-            .interestTags(RichiSkills)
-            .projectIDs(projectIDs)
-            .buildPerson();
-    people.add(Richi);
+    datastore = DatastoreServiceFactory.getDatastoreService();
   }
 
   @Override
   // get a specific user. return null if not found. TODO: User Validation
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userString = request.getParameter("user");
-
-    PersonObject result =
-        people.stream()
-            .filter(person -> userString.equals(person.getGitHubID()))
-            .findAny()
-            .orElse(null);
-    if (result == null) {
+    String userGithub = request.getParameter("githubID");
+    Key userKey = KeyFactory.createKey(PersonObject.ENTITY_NAME, userGithub);
+
+    PersonObject userObject;
+
+    try {
+      userObject = toPersonObject(datastore.get(userKey));
+    } catch (Exception e) {
       sendRawTextError(
-          response, HttpServletResponse.SC_BAD_REQUEST, "invalid user or user does not exist");
+          response, HttpServletResponse.SC_BAD_REQUEST, "Person does not exist in Datastore");
+      userObject = null;
     }
 
-    String jsonPerson = new Gson().toJson(result);
+    String jsonPerson = new Gson().toJson(userObject);
     response.setContentType("application/json;");
     response.getWriter().println(jsonPerson);
   }
 
+  @Override
+  // get a specific user. return null if not found.
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String userGitHubID = request.getParameter("gitHubID");
+    if (userGitHubID.isBlank()) {
+      sendRawTextError(response, HttpServletResponse.SC_BAD_REQUEST, "Invalid ID");
+      return;
+    }
+    ArrayList<String> userTags =
+        (ArrayList<String>) Arrays.asList(request.getParameterValues("tags"));
+
+    sendData(new PersonBuilder().gitHubID(userGitHubID).interestTags(userTags).buildPerson());
+  }
+
+  public void sendData(PersonObject person) {
```

Comment:
This function name is a bit confusing imo. It makes me think the data is being sent to the client. Consider something like:
```suggestion
  public void storePersonObject(PersonObject person) {
```